records_mover/creds/base_creds.py

```diff
@@ -140,11 +147,48 @@ def default_gcs_client(self) -> Optional['google.cloud.storage.Client']:
             return self.__default_gcs_client
 
     def default_db_facts(self) -> DBFacts:
-        if self.__default_db_facts is not NotYetFetched.token:
+        if self.__default_db_facts is not PleaseInfer.token:
             return self.__default_db_facts
 
         if self._default_db_creds_name is None:
             self.__default_db_facts = db_facts_from_env()
         else:
             self.__default_db_facts = self.db_facts(self._default_db_creds_name)
         return self.__default_db_facts
+
+    def _infer_scratch_s3_url(self,
+                              boto3_session: Optional['boto3.session.Session']) -> Optional[str]:
+        if "SCRATCH_S3_URL" in os.environ:
+            return os.environ["SCRATCH_S3_URL"]
+
+        config_result = get_config('records_mover', 'bluelabs')
+        cfg = config_result.config
+        if 'aws' in cfg:
+            aws_cfg = cfg['aws']
+            s3_scratch_url: Optional[str] = aws_cfg.get('s3_scratch_url')
+            if s3_scratch_url is None:
+                s3_scratch_url_prefix: Optional[str] =\
+                    aws_cfg.get('s3_scratch_url_appended_with_iam_user_id')
+                if s3_scratch_url_prefix is not None:
+                    if boto3_session is None:
+                        logger.warning('Cannot generate S3 scratch URL with IAM user ID, '
+                                       'as I have no IAM user ID')
+                        return None
+                    # TODO: Make sure boto3_session is not None
+                    sts_client = boto3_session.client('sts')
+                    caller_identity = sts_client.get_caller_identity()
+                    arn = caller_identity['Arn']
+                    # TODO: What happens if nothing follows this?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker